vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/bitflyer/post_oauth/post_oauth_bitflyer.cc

```diff
@@ -0,0 +1,163 @@
+      ledger::bitflyer::GenerateRandomString(ledger::is_testing);
+  const std::string hashed_code_verifier =
+      crypto::SHA256HashString(code_verifier);
+  std::string code_challenge;
+  base::Base64UrlEncode(hashed_code_verifier,
+                        base::Base64UrlEncodePolicy::INCLUDE_PADDING,
+                        &code_challenge);
+  dict.SetStringKey("code_verifier", code_verifier);
+  dict.SetStringKey("code_challenge_method", "S256");
+  dict.SetStringKey("code_challenge", code_challenge);
+
+  std::string payload;
+  base::JSONWriter::Write(dict, &payload);
+  return payload;
+}
+
+type::Result PostOauth::CheckStatusCode(const int status_code) {
+  if (status_code == net::HTTP_UNAUTHORIZED) {
+    return type::Result::EXPIRED_TOKEN;
+  }
+
+  if (status_code != net::HTTP_OK) {
+    return type::Result::LEDGER_ERROR;
+  }
+
+  return type::Result::LEDGER_OK;
+}
+
+type::Result PostOauth::ParseBody(const std::string& body,
+                                  std::string* token,
+                                  std::string* address,
+                                  std::string* linking_info) {
+  DCHECK(token);
+  DCHECK(address);
+  DCHECK(linking_info);
+
+  base::Optional<base::Value> value = base::JSONReader::Read(body);
+  if (!value || !value->is_dict()) {
+    BLOG(0, "Invalid JSON");
+    return type::Result::LEDGER_ERROR;
+  }
+
+  base::DictionaryValue* dictionary = nullptr;
+  if (!value->GetAsDictionary(&dictionary)) {
+    BLOG(0, "Invalid JSON");
+    return type::Result::LEDGER_ERROR;
+  }
+
+  const auto* access_token = dictionary->FindStringKey("access_token");
+  if (!access_token) {
+    BLOG(0, "Missing access token");
+    return type::Result::LEDGER_ERROR;
+  }
+
+  const auto* deposit_id = dictionary->FindStringKey("deposit_id");
+  if (!deposit_id) {
+    BLOG(0, "Missing deposit id");
+    return type::Result::LEDGER_ERROR;
+  }
+
+  const auto* linking_information = dictionary->FindStringKey("linking_info");
+  if (!linking_information) {
+    BLOG(0, "Missing linking info");
+    return type::Result::LEDGER_ERROR;
+  }
+
+  *token = *access_token;
+  *address = *deposit_id;
+  *linking_info = *linking_information;
+
+  return type::Result::LEDGER_OK;
+}
+
+void PostOauth::Request(const std::string& external_account_id,
+                        const std::string& code,
+                        PostOauthCallback callback) {
+  auto url_callback = std::bind(&PostOauth::OnRequest, this, _1, callback);
+
+  auto request = type::UrlRequest::New();
+  request->url = GetUrl();
+  request->content = GeneratePayload(external_account_id, code);
+  request->headers = RequestAuthorization();
+  request->content_type = "application/json";
+  request->method = type::UrlMethod::POST;
+  ledger_->LoadURL(std::move(request), url_callback);
+}
+
+void PostOauth::OnRequest(const type::UrlResponse& response,
+                          PostOauthCallback callback) {
+  ledger::LogUrlResponse(__func__, response, true);
```

Comment:
This will log user's `access token` and `refresh token` in the log. We should mask the sensitive fields in the body before logging.

Reply:
The parameter `true` at the end is for `long_response`, which will end up causing the body to be logged at a higher log level: 7 instead of 6. Since only log levels 6 and below are written to the diagnostic log file (when enabled via command line args, of course), the end result is that this `LogUrlResponse` call won't write the response body to the log file.

It's pretty confusing though. @emerick what would you think about just removing this line?

Reply:
Yeah, agreed it's confusing. My gut instinct is to keep it, though, until we come up with a simpler approach. Maybe we can add a comment to the `LogUrlResponse` for now to at least explain how this works?